crates/sui-adapter/src/adapter.rs

```diff
@@ -1150,7 +1182,9 @@ fn check_shared_object_rules(
         .iter()
         .map(|(id, obj)| (*id, obj.borrow().owner))
         .collect();
-    let ancestor_map = ObjectRootAncestorMap::new(&object_owner_map)?;
+    let ancestor_map =
+        ObjectRootAncestorMap::new(&object_owner_map).expect("ownership should be well formed");
```

Comment:
This is unfortunately not guaranteed to always be Ok. `resolve_and_type_check` today can be called from the gateway, which may not always have all objects.